core/services/relay/evm/relayer_extender.go

```diff
@@ -64,12 +68,13 @@ func NewLegacyChains(ctx context.Context, opts legacyevm.ChainRelayOpts) (result
 		}
 
 		privOpts.Logger.Infow(fmt.Sprintf("Loading chain %s", cid), "evmChainID", cid)
-		chain, err2 := legacyevm.NewTOMLChain(ctx, enabled[i], privOpts)
+		chain, err2 := legacyevm.NewTOMLChain(ctx, enabled[i], privOpts, chainIDToClientMap)
 		if err2 != nil {
 			err = multierr.Combine(err, fmt.Errorf("failed to create chain %s: %w", cid, err2))
 			continue
 		}
 
+		chainIDToClientMap[cid] = chain.Client()
```

Comment:
This is fine for now, but for LOOPP support we will have to pass relayers instead of local clients. Just something to keep in mind.

Reply:
We won't be able to support the `ethereum.FeeHistory` type there, for example, but we can sort that out later since it looks like a simple/generic type anyways.

Reply:
what's LOOPP support ?

Reply:
LOOP Plugins. The chain-agnostic "relayer" API that sits in front of each chain family implementation. We'll have to update this for the EVM LOOP Plugin.

https://github.com/smartcontractkit/chainlink-common/blob/main/pkg/types/relayer.go#L97